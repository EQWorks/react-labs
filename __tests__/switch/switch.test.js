import React from 'react'
import { render } from '@testing-library/react'

import Switch from '../../src/switch'

describe('render', () => {
  it('should render ok with no props passed', () => {
    const { getByTestId } = render(<Switch />)
    const switchElement = getByTestId('switch')
    // snapshot
    expect(switchElement).toMatchSnapshot()
    // no checked
    expect(switchElement).not.toHaveClass('Mui-checked')
    // not disabled
    expect(switchElement).not.toHaveClass('Mui-disabled')
  })
})
