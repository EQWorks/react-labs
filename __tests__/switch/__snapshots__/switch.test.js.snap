// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render ok with no props passed 1`] = `
<span
  aria-disabled="false"
  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase makeStyles-switchBase-3 MuiSwitch-colorSecondary"
  data-testid="switch"
>
  <span
    class="MuiIconButton-label"
  >
    <input
      class="PrivateSwitchBase-input-12 MuiSwitch-input"
      type="checkbox"
      value=""
    />
    <span
      class="MuiSwitch-thumb makeStyles-thumb-4 makeStyles-thumb-8"
    />
  </span>
</span>
`;
